packages/eds-demo/src/app/_components/Typography.tsx

```diff
@@ -0,0 +1,235 @@
+import { ReactNode, HTMLAttributes, CSSProperties } from "react";
+
+type FontSize =
+  | "xs"
+  | "sm"
+  | "md"
+  | "lg"
+  | "xl"
+  | "2xl"
+  | "3xl"
+  | "4xl"
+  | "5xl"
+  | "6xl";
+type LineHeight = "normal" | "squished";
+type As =
+  | "p"
+  | "span"
+  | "div"
+  | "h1"
+  | "h2"
+  | "h3"
+  | "h4"
+  | "h5"
+  | "h6"
+  | "label"
+  | "strong"
+  | "em";
+
+interface TypographyProps
+  extends Omit<HTMLAttributes<HTMLElement>, "className"> {
+  children: ReactNode;
+  size?: FontSize;
+  lineHeight?: LineHeight;
+  baselineAligned?: boolean;
+  debug?: boolean;
+  as?: As;
+  className?: string;
+  weight?: "lighter" | "normal" | "bolder";
+}
+
+/**
+ * Typography component using the EDS 2.0 typography system
+ *
+ * Implements a data-attribute driven sizing system with:
+ * - Baseline grid text alignment
+ * - Space toggle based CSS custom properties
+ * - Ratio-based padding calculations for optical alignment
+ *
+ * @example
+ * // Basic usage
+ * <Typography size="md">Body text</Typography>
+ *
+ * @example
+ * // With all props
+ * <Typography
+ *   size="lg"
+ *   lineHeight="squished"
+ *   baselineAligned={false}
+ *   weight="semibold"
+ *   as="h2"
+ * >
+ *   Heading
+ * </Typography>
+ */
+export function Typography({
+  children,
+  size = "md",
+  lineHeight = "normal",
+  baselineAligned = true,
```

Comment:
I like the convention where we prefix booleans with is or has. What do you think @pomfrida?